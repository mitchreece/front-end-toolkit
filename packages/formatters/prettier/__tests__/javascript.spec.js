const { resolve } = require('path');
const { getFileContent, formatFile } = require('./utils/file');

const filePath = resolve(__dirname, './samples/sample.js');

describe('JavaScript formatting', () => {
    test('matches snapshot', () => {
        const fileContents = getFileContent(filePath);
        const formattedFile = formatFile(fileContents, { parser: 'babel' });

        expect(formattedFile).toMatchSnapshot();
    });
});
