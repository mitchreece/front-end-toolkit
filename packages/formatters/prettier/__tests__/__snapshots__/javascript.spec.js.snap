// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JavaScript formatting matches snapshot 1`] = `
"const Country = Object.freeze({
  GB: 'Great Britain',
  FR: 'France',
  AU: 'Australia',
  US: 'United States',
})

const getCountryCode = (country) => {
  const countryMatch = Object.values(Country).indexOf(country)

  return {
    name: country,
    ...(countryMatch && {
      code: Object.keys(Country)[countryMatch],
    }),
  }
}

const countries = ['Great Britian', 'France', 'United States']

module.exports = countries.map(getCountryCode)
"
`;
